OpenEdXMobile/src/main/java/org/edx/mobile/module/analytics/Analytics.java

```diff
@@ -551,9 +551,79 @@ void trackValuePropLearnMoreTapped(@NonNull String courseId, @Nullable String as
+
+    /**
+     * Track if the application has the read/write access of calendar or not
+     *
+     * @param courseId      Id of the course
+     * @param userType      Whether the Account type of the user is audit or verified
+     * @param isSelfPaced   Whether the course is self-paced or instructor paced
+     * @param hasPermission True if the application has permission of calendar otherwise False
+     */
+    void trackCourseDatesCalendarPermission(@NonNull String courseId, @NonNull String userType, @NonNull boolean isSelfPaced, boolean hasPermission);
+
+    /**
+     * Track whether the user confirm to allow the events to be added on calendar or not
+     *
+     * @param courseId    Id of the course
+     * @param userType    Whether the Account type of the user is audit or verified
+     * @param isSelfPaced Whether the course is self-paced or instructor paced
+     * @param canAddDates True if the user confirmed to allow the events to be added on calendar otherwise False
+     */
+    void trackCourseDatesCalendarAddDates(@NonNull String courseId, @NonNull String userType, @NonNull boolean isSelfPaced, boolean canAddDates);
+
+    /**
+     * Track if the events added on calendar successfully
+     *
+     * @param courseId    Id of the course
+     * @param userType    Whether the Account type of the user is audit or verified
+     * @param isSelfPaced Whether the course is self-paced or instructor paced
+     */
+    void trackCourseDatesCalendarAddedSuccessfully(@NonNull String courseId, @NonNull String userType, @NonNull boolean isSelfPaced);
+
+    /**
+     * Track if the events removed from calendar successfully
+     *
+     * @param courseId    Id of the course
+     * @param userType    Whether the Account type of the user is audit or verified
+     * @param isSelfPaced Whether the course is self-paced or instructor paced
+     */
+    void trackCourseDatesCalendarRemovedSuccessfully(@NonNull String courseId, @NonNull String userType, @NonNull boolean isSelfPaced);
+
+    /**
+     * Track the user tapped on ViewEvents on Success dialog of the events added on calendar
+     *
+     * @param courseId    Id of the course
+     * @param userType    Whether the Account type of the user is audit or verified
+     * @param isSelfPaced Whether the course is self-paced or instructor paced
+     */
+    void trackCourseDatesViewCalendar(@NonNull String courseId, @NonNull String userType, @NonNull boolean isSelfPaced);
+
+    /**
+     * Track the user tapped on Done on Success dialog of the events added on calendar
+     *
+     * @param courseId    Id of the course
+     * @param userType    Whether the Account type of the user is audit or verified
+     * @param isSelfPaced Whether the course is self-paced or instructor paced
+     */
+    void trackCourseDatesCalendarConfirmSuccess(@NonNull String courseId, @NonNull String userType, @NonNull boolean isSelfPaced);
```

Comment:
IMO, we have to create only one method to track calendar events name `trackCalendarEvent`. cuz all the param are the same in calendar events.

Reply:
updated